e2e/main.e2e.spec.ts

```diff
@@ -40,69 +43,55 @@ test.describe.serial('Main App Test', () => {
   });
 
   test('should land on General page', async() => {
-    mainTitle = page.locator(mainTitleSelector);
+    const navPage = new NavPage(page);
 
-    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+    await expect(navPage.mainTitleSelector).toHaveText('Welcome to Rancher Desktop');
   });
 
   test('should start loading the background services and hide progress bar', async() => {
-    const progressBarSelector = page.locator('.progress');
+    const navPage = new NavPage(page);
 
-    // Wait until progress bar show up. It takes roughly ~60s to start in CI
-    await progressBarSelector.waitFor({ state: 'visible', timeout: 200_000 });
-    // Wait until progress bar be detached. With that we can make sure the services were started
-    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
-    await expect(progressBarSelector).toBeHidden();
+    await navPage.progressBecomesReady();
+    await expect(navPage.progressBarSelector).toBeHidden();
   });
 
   test('should navigate to Kubernetes Settings and check elements', async() => {
-    const k8sMemorySliderSelector = '[id="memoryInGBWrapper"]';
-    const k8sCpuSliderSelector = '[id="numCPUWrapper"]';
-    const k8sPortSelector = '[data-test="portConfig"]';
-    const k8sResetBtn = '[data-test="k8sResetBtn"]';
-
-    await navigateTo('K8s');
-    // Collecting data from selectors
-    const k8sSettingsTitle = page.locator(mainTitleSelector);
-    const k8sMemorySlider = page.locator(k8sMemorySliderSelector);
-    const k8sCpuSlider = page.locator(k8sCpuSliderSelector);
-    const k8sPort = page.locator(k8sPortSelector);
-    const k8sResetButton = page.locator(k8sResetBtn);
+    const navPage = new NavPage(page);
+    const k8sPage = new K8sPage(page);
+
+    await navPage.navigateTo('K8s');
 
     if (!os.platform().startsWith('win')) {
-      await expect(k8sMemorySlider).toBeVisible();
-      await expect(k8sCpuSlider).toBeVisible();
+      await expect(k8sPage.k8sMemorySliderSelector).toBeVisible();
+      await expect(k8sPage.k8sCpuSliderSelector).toBeVisible();
```

Comment:
Hmm, shouldn't we be testing that they're _not_ visible on Windows?

Reply:
Totally! nice catch :) 
I'll fix that